examples/webgl_shadow_contact.html

```diff
@@ -0,0 +1,279 @@
+				darkness: 1,
+				opacity: 1,
+				showWireframe: false,
+			};
+
+			var shadowGroup, renderTarget, renderTargetBlur, shadowCamera, cameraHelper, depthMaterial, horizontalBlurMaterial, verticalBlurMaterial;
+
+			var planeGeometry, planeMaterial, plane, blurPlane;
+
+			init();
+			animate();
+
+			function init() {
+
+				camera = new THREE.PerspectiveCamera( 50, window.innerWidth / window.innerHeight, 0.1, 100 );
+				camera.position.set( 0.5, 1, 2 );
+
+				scene = new THREE.Scene();
+
+				stats = new Stats();
+				document.body.appendChild( stats.dom );
+
+				window.addEventListener( 'resize', onWindowResize );
+
+				// add the example meshes
+
+				var geometries = [
+					new THREE.BoxBufferGeometry( 0.4, 0.4, 0.4 ),
+					new THREE.IcosahedronBufferGeometry( 0.3 ),
+					new THREE.TorusKnotBufferGeometry( 0.4, 0.05, 256, 24, 1, 3 )
+				];
+
+				var material = new THREE.MeshNormalMaterial();
+
+				for ( var i = 0, l = geometries.length; i < l; i ++ ) {
+
+					var angle = ( i / l ) * Math.PI * 2;
+
+					var geometry = geometries[ i ];
+					var mesh = new THREE.Mesh( geometry, material );
+					mesh.position.y = 0.1;
+					mesh.position.x = Math.cos( angle ) / 2.0;
+					mesh.position.z = Math.sin( angle ) / 2.0;
+					scene.add( mesh );
+					meshes.push( mesh );
+
+				}
+
+
+
+				// the container, if you need to move the plane just move this
+				shadowGroup = new THREE.Group();
+				shadowGroup.position.y = - 0.3;
+				scene.add( shadowGroup );
+
+				// the render target that will show the shadows in the plane texture
+				renderTarget = new THREE.WebGLRenderTarget( 512, 512 );
+				renderTarget.texture.generateMipmaps = false;
+
+				// the render target that we will use to blur the first render target
+				renderTargetBlur = new THREE.WebGLRenderTarget( 512, 512 );
+				renderTargetBlur.texture.generateMipmaps = false;
+
+
+				// make a plane and make it face up
+				planeGeometry = new THREE.PlaneGeometry( PLANE_WIDTH, PLANE_HEIGHT ).rotateX( Math.PI / 2 );
+				planeMaterial = new THREE.MeshBasicMaterial( {
+					map: renderTarget.texture,
+					opacity: state.opacity,
+					transparent: true,
+				} );
+				plane = new THREE.Mesh( planeGeometry, planeMaterial );
+				shadowGroup.add( plane );
+
+				// the y from the texture is flipped!
+				plane.scale.y = - 1;
+
+				// the plane onto which to blur the texture
+				blurPlane = new THREE.Mesh( planeGeometry );
+				blurPlane.visible = false;
+				shadowGroup.add( blurPlane );
+
+				// the camera to render the depth material from
+				shadowCamera = new THREE.OrthographicCamera( - PLANE_WIDTH / 2, PLANE_WIDTH / 2, PLANE_HEIGHT / 2, - PLANE_HEIGHT / 2, 0, CAMERA_HEIGHT );
+				shadowCamera.rotation.x = Math.PI / 2; // get the camera to look up
+				shadowGroup.add( shadowCamera );
+
+				cameraHelper = new THREE.CameraHelper( shadowCamera );
+
+				// like MeshDepthMaterial, but goes from black to transparent
+				depthMaterial = new THREE.ShaderMaterial( {
```

Comment:
Could you also use `MeshDepthMaterial` here and perform the changes with `onBeforeCompile()`?